### PR TITLE
Courses Page Fixes

### PR DESCRIPTION
**Pull request by ngwattcos**
#### Behavior Changes & Bug Fixes
* fixed a bug where "Home" and "Cancel" buttons kept flashing/disappearing in Edit mode on save
* changed to only display courses that user is enrolled in for "/home"; "/edit" still displays all available courses
* saving changes redirects to /home instead of the most recently visited course (especially since user may deselect al courses)
* changed workflow to allow all users to access "Edit Courses" page. Access to other courses should not take into account role (student, TA, prof) as this could increase surface area for bugs.

#### Implementation Changes
* hookify some state in CourseSelection view (isWritingChanges, pageState)
* replace "isRedirectingRef" to simple state hook "isWritingChanges"
* changed misnomer variable "canSave" to "isSaveDisabled" state hook
* disabled suppressing component re-render to allow useEffect to correctly update "isSaveDisabled" state for buttons (since it should depend on isWritingChanges hook)
* code beginnings of page state (ready, pending states). May need to move enum type to separate type declaration file

### Test Plan 
HOW TO TEST: Need to test "Edit Courses" and "Home" pages with student, TA, and professor types. Need to test this page with brand new user (and thus engaging isNormalEditingMode) via onboarding and as an existing user with ≥ 1 course loaded via the CalendarHeader dropdown. Try all combinations of these lol.

Tested with "existing-user professor" and "new-user student user" types. 

### Breaking Changes 

None

- [ ] I have updated the documentation accordingly.
- [ ] My PR adds a @ts-ignore

### CODE REVIEW

File: src/components/includes/CourseSelection.tsx [hunk 1]
```diff
@@ -13,19 +13,28 @@ type Props = {
     readonly isEdit: boolean;
 };
 
+export enum PageState {
+    ready = 'ready',
+    pending = 'pending',
+}
```
Comment (SamChou19815) [inline src/components/includes/CourseSelection.tsx]: Enum has a runtime cost. Try using literal types instead.
```ts
type PageState = 'ready' | 'pending'
```

File: src/styles/index.min.css [hunk 2]
```diff
@@ -1 +1 @@
...border:solid 1px #484848;margin:5px}.EnrollBar .buttons .disabled{opacity:.7;cursor:not-allowed}@media only screen and (max-width:920px){.App{background-color:white}.CourseSelection{background-color:white}.CourseSelection .QMeLogo.course{display:none}.CourseSelection .topBar{display:none}.CourseSelection .selectionContent{margin-top:0px}.CourseSelection .selectionContent .description{padding-bottom:25px;border-bottom:solid 2px #e2e2e2}.CourseSelection .selectionContent .description .title{font-size:24px;padding-bottom:20px;font-weight:bold}.CourseSelection .selectionContent .description .subtitle{font-size:18px}.CourseSelection .selectionContent .description .subtitle .EnrolledCourses.mobile{display:none}.EnrollBar{background-color:white;justify-content:space-around}.EnrollBar .buttons{max-width:500px}.EnrollBar .EnrolledCourses.web{display:none}}#root{height:100%}.App{text-align:center;height:100%}body{margin:0;padding:0;font-family:'Roboto','Lato','Helvetica Neue',Arial,Helvetica,sans-serif !important;background:#f5f5f5 !important}ul{color:#747474;list-style-type:none;margin:0;padding:0;text-align:left}ul li{border-bottom:solid 1px #747474;padding:5px 0}nav{background-color:#eee}
```
Comment (SamChou19815) [inline src/styles/index.min.css]: don't commit this generated file! Merge in my latest gitignore fix.

Comment (ngwattcos) [reply src/styles/index.min.css]: How do I merge this change?  Do I pull from master into this branch?

File: src/components/includes/CourseSelection.tsx (hunk 1 above)
Comment (ngwattcos) [reply src/components/includes/CourseSelection.tsx]: Interesting... good to know!

File: src/styles/index.min.css (hunk 2 above)
Comment (SamChou19815) [reply src/styles/index.min.css]: For this diff, just delete this file and never commit it again. In the future, you should start your branch against latest master.

Comment (ngwattcos) [reply src/styles/index.min.css]: Ok got it

File: src/components/includes/CourseSelection.tsx [hunk 3]
```diff
@@ -13,19 +13,25 @@ type Props = {
     readonly isEdit: boolean;
 };
 
+export type PageState = 'ready' | 'pending';
+
 function CourseSelection({ user, isEdit, allCourses }: Props): React.ReactElement {
     const history = useHistory();
-    const isRedirectingRef = React.useRef(false);
+    const [isWritingChanges, setIsWritingChanges] = React.useState(false);
+    const [pageState, setPageState] = React.useState('ready');
```
Comment (SamChou19815) [inline src/components/includes/CourseSelection.tsx]: `pageState` seems to be unused. See linter warnings in `Files changed`

Comment (SamChou19815) [reply src/components/includes/CourseSelection.tsx]: @ngwattcos is this state even useful

Comment (ngwattcos) [reply src/components/includes/CourseSelection.tsx]: At the moment, no. Can totally delete it, but I was basing this on your suggestion for page component state type. This is a first step to allow the possibility for conditionally rendering loading icons on any component.

**Pull request merged**